clang/lib/Basic/Targets/X86.cpp

```diff
@@ -130,17 +131,35 @@ bool X86TargetInfo::initFeatureMap(
       continue;
     }
 
-    if (!HasAVX512F && Feature.substr(0, 7) == "+avx512")
+    if (Feature.substr(0, 7) == "+avx10.") {
+      HasAVX10 = true;
       HasAVX512F = true;
-    if (HasAVX512F && Feature == "-avx512f")
+      if (Feature.substr(Feature.size() - 3, 3) == "512") {
+        HasEVEX512 = true;
+      } else if (Feature.substr(7, 2) == "1-") {
+        HasEVEX512 = false;
+      }
+    } else if (!HasAVX512F && Feature.substr(0, 7) == "+avx512") {
+      HasAVX512F = true;
+    } else if (HasAVX512F && Feature == "-avx512f") {
+      HasAVX512F = false;
+    } else if (HasAVX10 && Feature == "-avx10.1-256") {
+      HasAVX10 = false;
       HasAVX512F = false;
-    if (HasEVEX512 && Feature == "-evex512")
+    } else if (!HasEVEX512 && Feature == "+evex512") {
+      HasEVEX512 = true;
+    } else if (HasEVEX512 && Feature == "-avx10.1-512") {
```

Comment:
`+avx10.1-512,-avx10.1-512` effectively means `+avx10.1-256`?

Generally, when I see such flag dangling, I'm used to thinking there is likely to be a bug. Do we have some reasoning that it is correct? Or maybe it's just my internal bias.

Reply:
Yes and no.
We have already filtered this case (and cases like this in prior features) in the driver. Only the last argument will be passed to Clang if there are both positive and negative set in command line. So it won't turn into `+avx10.1-256`.
Function attrribute is another story, we seldom use negative features in function attrribute, and if we use, yes, it will enable features it implies. I think it makes sense that the attrribute turns off itself rather than features command line enabled. See a precedent https://godbolt.org/z/xKWfvEWY7